web/src/Sidebar.tsx

```diff
@@ -33,24 +47,134 @@ class SidebarItem {
    * Create a pared down SidebarItem from a ResourceView
    */
   constructor(res: Resource) {
+    let buildHistory = res.buildHistory || []
+    let lastBuild = buildHistory.length > 0 ? buildHistory[0] : null
+
     this.name = res.name ?? ""
     this.isTiltfile = !!res.isTiltfile
     this.status = combinedStatus(res)
     this.hasEndpoints = (res.endpoints || []).length > 0
+    this.lastBuildDur =
+      lastBuild && lastBuild.startTime && lastBuild.finishTime
+        ? timeDiff(lastBuild.startTime, lastBuild.finishTime)
+        : null
     this.lastDeployTime = res.lastDeployTime ?? ""
     this.pendingBuildSince = res.pendingBuildSince ?? ""
     this.currentBuildStartTime = res.currentBuild?.startTime ?? ""
     this.alertCount = numberOfAlerts(res)
     this.triggerMode = res.triggerMode ?? TriggerMode.TriggerModeAuto
     this.hasPendingChanges = !!res.hasPendingChanges
     this.queued = !!res.queued
-    let buildHistory = res.buildHistory || []
-    if (buildHistory.length > 0) {
-      this.lastBuild = buildHistory[0]
-    }
+    this.lastBuild = lastBuild
   }
 }
 
+const barberpole = keyframes`
```

Comment:
👀 what's all this CSS doing in a .tsx file?

Reply:
Haha, whenever I write styling code I try to do so using [styled-components](https://styled-components.com/). I'm aware that this decision / guideline is not actually documented anywhere. Maybe it's time for that style guide to be written...